### PR TITLE
Rename BackendStoreDrivers

### PR DESCRIPTION
**Pull request by derekbit**
https://github.com/longhorn/longhorn/issues/6126

### CODE REVIEW

File: examples/v2/storageclass.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-  name: longhorn-spdk
+  name: longhorn-v2-data-engine
 provisioner: driver.longhorn.io
 allowVolumeExpansion: true
 reclaimPolicy: Delete
 volumeBindingMode: Immediate
 parameters:
   # backup, backingImage and snapshot related parameters are not supported in SPDK preview stage
-  numberOfReplicas: "2"
+  numberOfReplicas: "3"
```
Comment (innobead) [inline examples/v2/storageclass.yaml:11]: Please also help update https://github.com/longhorn/longhorn/blob/master/examples/storageclass.yaml

**Review by innobead — CHANGES_REQUESTED** [review]
LGTM, just one comment.

File: examples/v2/storageclass.yaml (hunk 1 above)
Comment (derekbit) [reply examples/v2/storageclass.yaml:11]: It is for legacy volumes. Need to update as well?

Comment (innobead) [reply examples/v2/storageclass.yaml:11]: yeah. just make it consistent.

Comment (derekbit) [reply examples/v2/storageclass.yaml:11]: Renamed.

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@mergify backport v1.5.x

**Pull request merged**